tests-ui/tests/store/modelToNodeStore.test.ts

```diff
@@ -0,0 +1,245 @@
+import { createPinia, setActivePinia } from 'pinia'
+import { beforeEach, describe, expect, it } from 'vitest'
+
+import {
+  ModelNodeProvider,
+  useModelToNodeStore
+} from '@/stores/modelToNodeStore'
+import { type ComfyNodeDefImpl, useNodeDefStore } from '@/stores/nodeDefStore'
+
+describe('useModelToNodeStore', () => {
+  let store: ReturnType<typeof useModelToNodeStore>
+  let nodeDefStore: ReturnType<typeof useNodeDefStore>
+
+  // Create minimal mock for testing - only includes 'name' field since that's
+  // the only property ModelNodeProvider constructor uses and tests verify
+  const createMockNodeDef = (name: string): ComfyNodeDefImpl => {
+    return { name } as ComfyNodeDefImpl
+  }
+
+  beforeEach(() => {
+    setActivePinia(createPinia())
+    store = useModelToNodeStore()
+    nodeDefStore = useNodeDefStore()
+
+    const mockNodeNames = [
+      'CheckpointLoaderSimple',
+      'ImageOnlyCheckpointLoader',
+      'LoraLoader',
+      'LoraLoaderModelOnly',
+      'VAELoader',
+      'ControlNetLoader',
+      'UNETLoader',
+      'UpscaleModelLoader',
+      'StyleModelLoader',
+      'GLIGENLoader'
+    ]
+
+    const mockNodeDefs: Record<string, ComfyNodeDefImpl> = Object.fromEntries(
+      mockNodeNames.map((name) => [name, createMockNodeDef(name)])
+    )
+
+    nodeDefStore.nodeDefsByName = mockNodeDefs
+  })
+
+  describe('modelToNodeMap', () => {
+    it('should initialize as empty', () => {
+      expect(Object.keys(store.modelToNodeMap)).toHaveLength(0)
+    })
+
+    it('should populate after registration', () => {
+      store.registerDefaults()
+      expect(Object.keys(store.modelToNodeMap)).toContain('checkpoints')
+      expect(Object.keys(store.modelToNodeMap)).toContain('unet')
+    })
+  })
+
+  describe('getNodeProvider', () => {
+    it('should return provider for registered model type', () => {
+      store.registerDefaults()
+
+      const provider = store.getNodeProvider('checkpoints')
+      expect(provider).toBeDefined()
+      // Optional chaining used because getNodeProvider() can return undefined for unregistered types
+      expect(provider?.nodeDef.name).toBe('CheckpointLoaderSimple')
```

Comment:
If one piece is possibly undefined, the chain should usually go to the terminus.
Otherwise you'll get that `node` isn't present on `undefined`.